### PR TITLE
Migrate to GitHub Actions.

### PR DESCRIPTION
**Pull request by jezdez**
Travis CI has a new pricing model which places limits on open source.

* https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
* https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

Many projects are moving to GitHub Actions instead, including Jazzband projects: 

* https://github.com/orgs/jazzband/projects/1
* https://github.com/jazzband-roadies/help/issues/206

This is based on https://github.com/jazzband/contextlib2/pull/26.

TODO:

* [x] @jezdez to add `JAZZBAND_RELEASE_KEY` to the repo secrets.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -14,15 +14,15 @@ django-defender
...
+.. image:: https://github.com/jazzband/django-defender/workflows/Test/badge.svg
+   :target: https://github.com/jazzband/django-defender/actions
+   :alt: GitHub Actions
 
 .. image:: https://codecov.io/gh/jazzband/django-defender/branch/master/graph/badge.svg
    :target: https://codecov.io/gh/jazzband/django-defender
    :alt: Coverage
 
-.. image:: https://readthedocs.org/projects/python-dockerflow/badge/?version=latest
+.. image:: https://readthedocs.org/projects/python-defender/badge/?version=latest
```
Comment (aleksihakli) [inline README.rst:25]: Should this be `django-defender`? `python-defender` is non-existent.

File: tox.ini [hunk 2]
```diff
@@ -15,14 +15,15 @@ python =
 [testenv]
```
Comment (aleksihakli) [inline tox.ini:15]: `[testenv:py{36,37,38,39,py3}-dj{master}]` needs to be excluded from the test matrix that is required to pass. At the moment installing `coverage` fails on Django master branch and that causes the whole test matrix to fail.

The following snippet should work for that:

```
[testenv:py{36,37,38,39,py3}-dj{master}]
ignore_errors = true
ignore_outcome = true
```

**Review by aleksihakli — CHANGES_REQUESTED** [review]
A few notes on the test matrix configuration and badges, looks good aside from those.

File: README.rst (hunk 1 above)
Comment (jezdez) [reply README.rst:25]: Typo in a typo!
```suggestion
.. image:: https://readthedocs.org/projects/django-defender/badge/?version=latest
```

File: tox.ini [hunk 3]
```diff
@@ -15,14 +15,15 @@ python =
...
+    dj31: django>=3.1,<3.2
+    djmaster: https://github.com/django/django/archive/master.tar.gz
 usedevelop = True
 commands =
-    {envbindir}/coverage run --source=defender {envbindir}/django-admin test defender --settings=defender.travis_settings
+    {envbindir}/coverage run --source=defender {envbindir}/django-admin test defender --settings=defender.ci_settings
     {envbindir}/coverage report -m
+    {envbindir}/coverage xml
 
 [testenv:py38-docs]
```
Comment (jezdez) [inline tox.ini:28]:
```suggestion
[testenv:py{36,37,38,39,py3}-dj{master}]
ignore_errors = true
ignore_outcome = true

[testenv:py38-docs]
```

File: tox.ini (hunk 2 above)
Comment (jezdez) [reply tox.ini:15]: Good catch!

**Review by aleksihakli — APPROVED** [review]

**Comment by aleksihakli** [comment]
Reviewed and LGTM 👍

**Comment by kencochrane** [comment]
Thank you @jezdez @aleksihakli

**Pull request merged**